### PR TITLE
  (IMAGES-1217) Fix double interface ip on ubuntu-1804+ 

### PR DESCRIPTION
**Pull request by sbeaulie**
ubuntu '18.04', '18.10', '20.04' includes a /etc/netplan that already configures the network with dhcp
Adding a network::interface sets up two interfaces each with an IP and creates issues with DNs resolution
and dhcp.

### CODE REVIEW

**Comment by sbeaulie** [comment]
needs a squash merge

**Comment by sbeaulie** [comment]
Both built via the CI pipelines https://cinext-jenkinsmaster-sre-prod-1.delivery.puppetlabs.net/job/imaging_ubuntu-18.04_x86_64_vmware_vcenter_packer_build_packer/

the last step where it creates the PR had issues witht he template folder name, but I manually fixed the commits.

File: manifests/modules/packer/templates/vsphere/ubuntu.rb.erb [hunk 1]
```diff
@@ -42,15 +42,16 @@ puts '- Re-obtaining DHCP lease...'
...
 <% elsif ['18.04', '18.10', '20.04'].include? @operatingsystemrelease -%>
-# systemd-networkd's DHCP client uses /etc/machine-id instead of the link layer address (as dhclient does) to generate a client ID
-# /etc/machine-id will be the same on VMs cloned from the same template, so regenerate it:
-File.delete('/etc/machine-id')
-Kernel.system('/bin/systemd-machine-id-setup')
-Kernel.system('/bin/systemctl daemon-reload')
+# netplan uses machine-id or client-it as identifier by default, but we need a different value for
+# each VM, so configure netplan's dhcp to use mac addresses
+#open('/etc/netplan/01-netcfg.yaml', 'a') { |f|
+#  f.puts "      dhcp-identifier: mac"
```
Comment (ciprianbadescu) [inline manifests/modules/packer/templates/vsphere/ubuntu.rb.erb:48]: When using the new images we are unable to access servers running on GKE cluster (https://jenkins-platform.delivery.puppetlabs.net/view/puppet-agent/view/Acceptance%20Suites/view/CI%20Goalie/job/platform_puppet-agent_puppet-agent-integration-suite_daily-5.5.x/RMM_COMPONENT_TO_TEST_NAME=pxp_agent,SLAVE_LABEL=beaker,TEST_TARGET=ubuntu1804-64a/841/console).

Digging into this we learned that the domain we are using (gke-puppetagent-ci.puppet.net) and which is an alias for "gke-puppetagent-ci.svc.cluster.local." will not be solved by systemd-resolved as ".local" has special meaning for it. We found a manageable workaround for it: to add domain 'local' to interface domain search list (systemd-resolve --set-domain local --interface ens192)
To do this directly in netplan configuration file, we need to add nameserver domain option like bellow: 
    ens192:
      dhcp4: yes
      dhcp-identifier: mac
      nameservers:
        search: [local]

Comment (sbeaulie) [reply manifests/modules/packer/templates/vsphere/ubuntu.rb.erb:48]: I tried this in floaty, to enable I needed to do netplan apply, as /bin/systemctl restart systemd-networkd did not pick up the change

**Pull request merged**